components/image-builder-mk3/pkg/orchestrator/orchestrator.go

```diff
@@ -228,12 +228,18 @@ func (o *Orchestrator) Build(req *protocol.BuildRequest, resp protocol.ImageBuil
 
 	// resolve build request authentication
 	reqauth := o.AuthResolver.ResolveRequestAuth(req.Auth)
+
+	// resolve to ref to baseImageNameResolved (if it exists)
 	if req.BaseImageNameResolved != "" && !req.GetForceRebuild() {
+		// allow access to baseImage repository so we can look it up later
+		req.Auth.GetSelective().AllowBaserep = true
```

Comment:
TIL about BuildRegistryAuthSelective

Reply:
Related: https://github.com/gitpod-io/gitpod/blob/aa58596eb4bc079da3c836dfef3eacf8f18e09d8/components/server/src/workspace/workspace-starter.ts#L1074-L1076

Reply:
BTW just pushed to change it to this:
```
		if req.Auth.GetSelective() != nil {
			// allow access to baseImage repository so we can look it up later
			req.Auth.GetSelective().AllowBaserep = true
			reqauth = o.AuthResolver.ResolveRequestAuth(req.Auth)
		}
```
to avoid the lurking nil deref :see_no_evil:

Reply:
Regarding https://github.com/gitpod-io/gitpod/pull/19745#discussion_r1601815771: this is the self-hosted feature I mentioned in the PR description:

> installation defined registry auth ([code](https://github.com/gitpod-io/gitpod/pull/19474/files#diff-b1c591e18af598ccba93b570ce7c97dece0988c9109a6d3be82bbe0df1fbd619L1057-L1062))
>  - this is an old self-hosted feature, and we don't seem to be using it anymore ✔️

Reply:
I am not seeing the push.

Instead of force pushing, can you push a new commit? Makes it easier to see differences, we can squash later.

Reply:
req.Auth can be nil, too (perhaps less likely, but still)

Reply:
Done